src/groupByKey.ts

```diff
@@ -1,7 +1,20 @@
+interface Item {
+  [key: string]: any;
+}
+
 type GroupsMap<T> = {
-  [key: string]: T[];
+  [key:string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: Item[], key: keyof Item):GroupsMap<Item> {
+  const groups: GroupsMap<Item> = {};
+
+  items.forEach((obj) => {
+    if (!Object.prototype.hasOwnProperty.call(groups, obj[key])) {
```

Comment:
let's just check if this key is in object (use in operator or object[key]), I think Object.prototype.hasOwnProperty.call looks strange)